### PR TITLE
arch-vega, dev-amdgpu: Fix for memory leaks

### PR DESCRIPTION
**Pull request by BujSet**
When using the new operator, delete should be called
on any allocated memory after it's use is complete.

Change-Id: Id5fcfb264b6ddc252c0a9dcafc2d3b020f7b5019

### CODE REVIEW

**Review by BobbyRBruce — APPROVED** [review]

**Review by powerjg — CHANGES_REQUESTED** [review]
Before merging, can you make sure to use your real name and "official" email address in the commit author info?

**Comment by mattsinc** [comment]
> When createRead is called, the new operator is used to allocate a packet. When the packet's use is complete, it should be freed (deleted).
> 
> Change-Id: Id5fcfb264b6ddc252c0a9dcafc2d3b020f7b5019

I think this part of the commit message is also a bit misleading.  With the PTWalker code, it does not appear callRead has anything to do with the leak there, right?

If so please update the commit message accordingly.

**Comment by BujSet** [comment]
> Before merging, can you make sure to use your real name and "official" email address in the commit author info?

@powerjg I believe I updated the commit to include my full name and email address, though I'm unable to verify if they are present on the latest commit on the pull request. Is the change visible to you?

**Comment by BobbyRBruce** [comment]
I'll leave this for @powerjg to give the final say on this. It need his review.

**Comment by powerjg** [comment]
When I run `git log` here's what I see:

```
commit 59c5cf5f7f4ac2d51ff34ae78778a21ede1f3618 (HEAD -> pr/BujSet/129, BujSet/memleaks)
Merge: a6839fe90b 4e3c22b51a
Author: Bujji <bujji.selagamsetty@amd.com>
Date:   Thu Jul 27 13:54:12 2023 -0500

    misc: Merge branch 'memleaks' into memleaks
    
    Change-Id: I4fb2ca77d802b87d92ea32b0d71a166e6ff465b9

commit a6839fe90b94e4be6a7a94eed448ef9fe51855b4
Author: Ranganath (Bujji) Selagamsetty <bujji.selagamsetty@amd.comcom>
Date:   Wed Jul 26 17:27:05 2023 -0500

    arch-vega, dev-amdgpu: Fix for memory leaks
    
    When using the new operator, delete should be called
    on any allocated memory after it's use is complete.
    
    Change-Id: Id5fcfb264b6ddc252c0a9dcafc2d3b020f7b5019

commit 4e3c22b51a9e2d5b54f39d83c5809402da13af9b
Merge: b6f1420444 65b99fffc9
Author: BujSet <ranganath1000@gmail.com>
Date:   Thu Jul 27 10:39:40 2023 -0700

    Merge branch 'develop' into memleaks

commit 65b99fffc93e3a7bddfffb4d3607e91c5a065779 (BujSet/develop)
Merge: 42b65cad68 f19945e9cb
Author: Bobby R. Bruce <bbruce@ucdavis.edu>
Date:   Thu Jul 27 09:39:18 2023 -0700

    util: Ignore line length check for #include pragma in C/C++ files (#134)
    
    The length of the path of #include pragmas can be more than
    79-character long. The length of the path of a #include pragma
    can be outside of user's control.
```

There's two things I'd still like to see improved:
1. Your email has a typo in the "main" commit (`amd.comcom`)
2. Can you rebase your changes to remove the extra merge commits that have differing `author` fields?

Thanks! Hopefully things are a bit smother next time!

**Comment by BujSet** [comment]
@powerjg I think the latest update should squash the extra logged commits and author updates. Can you verify?

**Review by powerjg — APPROVED** [review]
Perfect! Thank you!!!

**Review by abmerop — APPROVED** [review]

**Comment by BujSet** [comment]
Updated commit with latest changes from develop.

**Comment by BujSet** [comment]
Updated commit with latest changes from develop.

**Comment by BujSet** [comment]
Updated commit with latest changes from develop... again. Hopefully this time I can win the race to complete the regression tests before someone else pushes to develop. 🙏

**Comment by abmerop** [comment]
Correct me if I am wrong, but it shouldn't be a race. As long as it can apply cleanly on the tip of develop after the tests pass it should be mergeable

**Pull request merged**
